### PR TITLE
Remove logging

### PR DESCRIPTION
**Pull request by koaning**
Logging has now been removed from the population object. This caused a few changes: 

1. I found a small bug where the **kwargs weren't passed from the `Evolution` to the step. This was fixed. 
2. Logger is no longer passed to the population object but can be passed to the `.callback()` mechanic or called directly. 
3. The tests have been changed to reflect this change. 

There is a TODO to add a ExceptionType for `PopulationNotEvaluated`. This will be added later to prevent merge hardship.

### CODE REVIEW

**Comment by koaning** [comment]
It seems good to merge https://github.com/godatadriven/evol/pull/139 first since I've removed travis fucntionality in the meantime.

File: evol/step.py [hunk 1]
```diff
@@ -99,6 +99,7 @@ def __init__(self, name, every: int = 1, **kwargs):
         EvolutionStep.__init__(self, name, **kwargs)
         self.count = 0
         self.every = every
+        self.kwargs = kwargs
```
Comment (rogiervandergeer) [inline evol/step.py]: This already happens in the `__init__` of `EvolutionStep`.

File: evol/logger.py [hunk 2]
```diff
@@ -39,12 +39,19 @@ def __init__(self, target=None, stdout=False, fmt='%(asctime)s,%(message)s'):
                 self.logger.addHandler(stream_handler)
         self.logger.setLevel(level=logging.INFO)
 
+    @staticmethod
+    def check_population(population):
+        if any([i.fitness is None for i in population]):
```
Comment (rogiervandergeer) [inline evol/logger.py]: The islands PR contains an `is_evaluated` property on the population. That would be nice to use here.

File: evol/step.py (hunk 1 above)
Comment (koaning) [reply evol/step.py]: d0h.

File: evol/logger.py (hunk 2 above)
Comment (koaning) [reply evol/logger.py]: Sure. Is it easier to merge this and then the Islands or the other way around?

Comment (rogiervandergeer) [reply evol/logger.py]: As long as we fix it in the end 😊

Comment (koaning) [reply evol/logger.py]: agreed https://github.com/godatadriven/evol/issues/141

**Pull request merged**